lib/ansible/plugins/connection/ssh.py

```diff
@@ -253,6 +253,8 @@ def _send_initial_data(self, fh, in_data):
 
         try:
             fh.write(in_data)
+            if tty:
+                fh.write("__EOF__\n")
```

Comment:
I've pushed another commit to add a random suffix to reduce the probability of collisions. Thanks for the suggestion.